pkg/cloudencrypt/cache.go

```diff
@@ -0,0 +1,68 @@
+package cloudencrypt
+
+import (
+	"context"
+	"time"
+
+	"github.com/dgraph-io/ristretto/v2"
+)
+
+// CacheEncryptor wraps another Encryptor and adds a caching mechanism.
+type CacheEncryptor struct {
```

Comment:
This seems like we encrypt the cache entries instead of encrypting using cache

Reply:
How about `CachedEncryptor` and `LoggedEncryptor` then? :thinking: